### PR TITLE
Implement numpy-like Rayleigh distribution

### LINKED ISSUE #223: Implement Rayleigh RV as a scaled Chi2
### Description

Code:
```python
def rayleigh(scale=1.0, *, size=None, **kwargs):
  return chi2(df=2, size=size, **kwargs) * as_tensor_variable(scale)
```

from:
https://github.com/pymc-devs/pytensor/pull/220#issuecomment-1435947464

### PR DESCRIPTION
**Pull request by Dhruvanshu-Joshi**
## Description

Incorporate ```Rayleigh``` distribution as square root of ```chi2```  distribution with degree of freedom 2.
As per the [reference](https://stats.libretexts.org/Bookshelves/Probability_Theory/Probability_Mathematical_Statistics_and_Stochastic_Processes_(Siegrist)/05%3A_Special_Distributions/5.14%3A_The_Rayleigh_Distribution#:~:text=Connections%20to%20the,standard%20Rayleigh%20distribution.).

## Related Issue

- [x] Closes #223 
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/tensor/random/basic.py [hunk 1]
```diff
@@ -526,6 +527,13 @@ def chisquare(df, size=None, **kwargs):
     return gamma(shape=df / 2.0, scale=2.0, size=size, **kwargs)
 
 
+def rayleigh(scale=1.0, *, size=None, **kwargs):
+    scale = as_tensor_variable(scale)
+    if size is None:
+        size = np.broadcast_shapes(scale.shape)
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:533]:
```suggestion
        size = scale.shape
```

File: pytensor/tensor/random/basic.py [hunk 2]
```diff
@@ -526,6 +527,13 @@ def chisquare(df, size=None, **kwargs):
     return gamma(shape=df / 2.0, scale=2.0, size=size, **kwargs)
 
 
+def rayleigh(scale=1.0, *, size=None, **kwargs):
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:530]: Missing docstrings like other RandomVariables have (usually in `__call__` method)

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Looks good, just a small tweak and need for docstrings

File: pytensor/tensor/random/basic.py [hunk 3]
```diff
@@ -526,6 +527,38 @@ def chisquare(df, size=None, **kwargs):
...
+    ---------
+    `() -> ()`
+
+    Parameters
+    ----------
+    scale : float or array_like of floats, optional
+        Scale parameter of the distribution (positive). Default is 1.0.
+    size : int or tuple of ints, optional
+        Output shape. If the given shape is, e.g., `(m, n, k)`, then `m * n * k` samples
+        are drawn. Default is None, in which case a single value is returned.
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:553]: This is wrong. Is it what the other docstrings write? If not provided the size is implied by the shape of `scale`

Comment (Dhruvanshu-Joshi) [reply pytensor/tensor/random/basic.py:553]: No this is a error on my side

**Review by ricardoV94 — APPROVED** [review]
Thanks

**Pull request merged**